components/Context/DdfInfo.js

```diff
@@ -24,9 +28,158 @@ export default class DdfInfo extends React.Component {
         loadingSize: "50"
       }],
       selectedCategory: "Development",
-      selectedFilter: "None",
+      categories: [],
+      showCompletedTasks: false,
+      setCategory: this.setCategory.bind(this),
+      handleClickedFilter: this.handleClickedFilter.bind(this),
+      handleShowCompletedTasks: this.handleShowCompletedTasks.bind(this),
     }
+
+    this.organizeIssues = this.organizeIssues.bind(this);
+  }
+
+  handleShowCompletedTasks(showCompletedTasks){
+    this.setState({showCompletedTasks})
+  }
+
+  handleClickedFilter(name, checked){
+    let tmpIssues = Object.assign({}, this.state.issues);
+    tmpIssues[this.state.selectedCategory].filters[name] = checked;
+    this.setState({issues: tmpIssues});
+  }
+
+  setCategory(selectedCategory){
+    if(this.state.selectedCategory !== selectedCategory){
+      this.setState({selectedCategory})
+    }
+  }
+
+ /*
+ * this method splits issues by categories and also sets it up so that
+ + its easy to handle the state of each filter
+ */
+ organizeIssues(issues){
+    let processedIssues = {};
+    let categories = [];
+
+    //isues that don't have a bounty
+    issues.filter(issue => issue.contractAddress === -1);
+
+    issues.forEach(issue => {
+      const category = issue.category;
+      if(!processedIssues[category]){
+        processedIssues[category] = {issues: [issue], filters:{}};
+        categories.push(category);
+      }
+      else{
+        processedIssues[category].issues.push(issue);
+      }
+      issue.labels.forEach(label => {
+        //if the user has toggled the filter off before then let it be
+        if(this.state.issues && this.state.issues[category] && this.state.isssues[category].filters[label] === false){
+          processedIssues[category].filters[label] = false;
+        }
+        else if(processedIssues[category].filters[label] === undefined){
+          //filters always start off as checked
+          processedIssues[category].filters[label] = true;
+        }
+      })
+    });
+
+    this.setState({issues: processedIssues, categories})
+ }
+
+ getContractAddress(issue, comments){
+  //TODO Change from owner to status' username once jose implements the end point
+  comments.filter(comment => comment.author_association === "OWNER");
+  let contractAddress = -1;
+  let match = comments[0].body.match(ethereumRegex());
+  if(match && match.length > 0){
+    contractAddress = match[0];
+  }
+  return {
+    ...issue,
+    contractAddress,
+    value: "$100"
+  };
+ }
+
+async getCommentsOfIssue(repoName, number){
+  return new Promise(async (resolve, reject) => {
+    axios.get(`https://api.github.com/repos/csmartinsfct/${repoName}/issues/${number}/comments?access_token=573dbede2e9766fd4d41610d15a8b82371fd522a`)
```

Comment:
Never commit in Github or any version control any sort of access token. I've mentioned to you a couple times to do a quick request on Postman and load the results in a separate json file so you can fetch and mock this instead.

This is untestable (results may vary), unmaintainable (hard to read), and overloaded (all logic embed in a component). Please:

* Separate all calls to our api into a `api` folder that performs specific calls against an endpoint.
* Store mock data into a `githubMockResponse.json` file and preload it on `componentDidMount`. Alternately you can use [nock](https://www.npmjs.com/package/nock) and detect `process.env.NODE_ENV` so you only nock on development mode.
* Write the tests for our api library w sinon

Then, and only then you can retrieve fetch this information. This code should look like:

```
import GithubApi from 'api/github';
...

async getCommentsOfIssue(repoName, number) =>  githubApi.getCommentsOfIssue(repoName, number)
```

Reply:
Will do.

Reply:
I've seen the other ones, let's halt this one for the time being.